stdlib/xml/dom/xmlbuilder.pyi

```diff
@@ -1,6 +1,109 @@
+_DOMBuilderErrorHandlerType = Optional[Any]
+# probably some kind of IO...
+_DOMInputSourceCharacterStreamType = Optional[Any]
+# probably a string??
+_DOMInputSourceStringDataType = Optional[Any]
+# probably a string??
+_DOMInputSourceEncodingType = Optional[Any]
+
+class Options:
+    namespaces: int
+    namespace_declarations: bool
+    validation: bool
+    external_parameter_entities: bool
+    external_general_entities: bool
+    external_dtd_subset: bool
+    validate_if_schema: bool
+    validate: bool
+    datatype_normalization: bool
+    create_entity_ref_nodes: bool
+    entities: bool
+    whitespace_in_element_content: bool
+    cdata_sections: bool
+    comments: bool
+    charset_overrides_xml_encoding: bool
+    infoset: bool
+    supported_mediatypes_only: bool
+    errorHandler: Any | None
+    filter: DOMBuilderFilter | None  # a guess, but seems likely
+
+class DOMBuilder:
+    entityResolver: DOMEntityResolver | None  # a guess, but seems likely
+    errorHandler: _DOMBuilderErrorHandlerType
+    filter: DOMBuilderFilter | None  # a guess, but seems likely
+    ACTION_REPLACE: Literal[1]
+    ACTION_APPEND_AS_CHILDREN: Literal[2]
+    ACTION_INSERT_AFTER: Literal[3]
+    ACTION_INSERT_BEFORE: Literal[4]
+    def __init__(self) -> None: ...
+    def setFeature(self, name: str, state: int) -> None: ...
+    def supportsFeature(self, name: str) -> bool: ...
+    def canSetFeature(self, name: str, state: int) -> bool: ...
+    # getFeature could return any attribute from an instance of `Options`
+    def getFeature(self, name: str) -> Any: ...
+    def parseURI(self, uri: str) -> ExpatBuilder | ExpatBuilderNS: ...
+    def parse(self, input: DOMInputSource) -> ExpatBuilder | ExpatBuilderNS: ...
+    # `input` and `cnode` argtypes for `parseWithContext` are unknowable
+    # as the function does nothing with them, and always raises an exception.
+    # But `input` is *probably* `DOMInputSource`?
+    def parseWithContext(self, input: object, cnode: object, action: Literal[1, 2, 3, 4]) -> NoReturn: ...
+
+class DOMEntityResolver:
+    def resolveEntity(self, publicId: str | None, systemId: str) -> DOMInputSource: ...
+
+class DOMInputSource:
+    byteStream: OpenerDirector | None
+    characterStream: _DOMInputSourceCharacterStreamType
+    stringData: _DOMInputSourceStringDataType
+    encoding: _DOMInputSourceEncodingType
+    publicId: str | None
+    systemId: str | None
+    baseURI: str | None
+
+class DOMBuilderFilter:
+    FILTER_ACCEPT: Literal[1]
+    FILTER_REJECT: Literal[2]
+    FILTER_SKIP: Literal[3]
+    FILTER_INTERRUPT: Literal[4]
+    whatToShow: int
+    # The argtypes for acceptNode and startContainer appear to be irrelevant.
+    def acceptNode(self, element: object) -> Literal[1]: ...
+    def startContainer(self, element: object) -> Literal[1]: ...
+
+class DocumentLS:
+    async_: bool
+    # The only requirements on the `flag` argument
+    # are that it can safely be passed to `bool()`.
```

Comment:
This comment no longer applies, because the method that had a `flag` argument was deleted.

Reply:
lol

Reply:
thanks!